### PR TITLE
Added :stack command

### PR DESCRIPTION
**Pull request by noughtmare**

### CODE REVIEW

File: yi-keymap-vim/src/Yi/Keymap/Vim/Ex/Commands/Stack.hs [hunk 1]
```diff
@@ -21,8 +24,46 @@ import           Yi.MiniBuffer                    (CommandArguments (CommandArgu
...
+    cmd <- "stack" *> (" " *> P.choice commands <|> "" $> "build")
+    args <- Common.commandArgs
     return $ Common.impureExCommand {
-        cmdShow = "stack build"
-      , cmdAction = YiA $ stackBuildE $ CommandArguments args
+        cmdShow = "stack commands (defaults to build)"
+      , cmdAction = YiA $ stackCommandE cmd $ CommandArguments args
       }
+
+commands :: [P.Parser T.Text]
```
Comment (jhance) [inline yi-keymap-vim/src/Yi/Keymap/Vim/Ex/Commands/Stack.hs]: Wat. Does `P.Parser` have an `IsString` instance?

File: yi-keymap-vim/src/Yi/Keymap/Vim/Ex/Commands/Stack.hs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TupleSections #-}
```
Comment (jhance) [inline yi-keymap-vim/src/Yi/Keymap/Vim/Ex/Commands/Stack.hs]: Where are you using tuple sections?

File: yi-keymap-vim/src/Yi/Keymap/Vim/Ex/Commands/Stack.hs (hunk 1 above)
Comment (noughtmare) [inline yi-keymap-vim/src/Yi/Keymap/Vim/Ex/Commands/Stack.hs]: [Yes](https://github.com/bos/attoparsec/blob/master/Data/Attoparsec/Text/Internal.hs#L94-L95)

Comment (jhance) [inline yi-keymap-vim/src/Yi/Keymap/Vim/Ex/Commands/Stack.hs]: Weird, I wonder why it didn't show up in the haddock docs for me

**Pull request merged**